src/main/java/com/users/demo_users/UserController.java

```diff
@@ -0,0 +1,33 @@
+package com.users.demo_users;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private final UserRepository userRepository;
+
+    @Autowired
+    public UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @PostMapping
+    public User addUser(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @GetMapping
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public User getUserById(@PathVariable Long id) {
+        return userRepository.findById(id).orElse(null);
```

Comment:
Use abstraction in controller. Use repository in class annotated with @Service, like UserService and let this class use your repository 

3 layers: https://medium.com/@deanrubin/the-three-layered-architecture-fe30cb0e4a6
You don't have to have it, but it's good for starters.

null is worth having exception. Because... user is not found.